autogen/oai/client_utils.py

```diff
@@ -0,0 +1,56 @@
+"""Utilities for client classes"""
+
+import warnings
+from typing import Any, Dict, List, Optional, Tuple
+
+
+def validate_parameter(
+    params: Dict[str, Any],
+    param_name: str,
+    allowed_types: Tuple,
+    allow_None: bool,
+    default_value: Any,
+    numerical_bound: Tuple,
+    allowed_values: list,
+) -> Any:
+    """Validates a given config parameter, checking its type, values, and setting defaults"""
+
+    if allowed_values is not None and not isinstance(allowed_values, list):
+        raise TypeError(f"allowed_values should be a list or None, got {type(allowed_values).__name__}")
+
+    param_value = params.get(param_name, default_value)
+    warning = ""
+
+    if param_value is None and allow_None:
+        pass
+    elif param_value is None and not allow_None:
+        warning = "cannot be None"
```

Comment:
Since check for param_values and check on numerical and allowed types is not exactly the same, I think it is clearer to separate them. Also, can numerical_bound and allowed_values be passed in at the same time?

```python
if param_value is None:
    if not allowed_None:
        warning = "cannot be None"
elif not isinstance(param_value, allowed_types):
...



if numerical_bound:
...
```